src/Parser/Extra.elm

```diff
@@ -0,0 +1,112 @@
+module Parser.Extra
+    exposing
+        ( between
+        , braces
+        , brackets
+        , by
+        , many
+        , parens
+        , some
+        , spacy
+        )
+
+{-| Some convenience parser combinators.
+
+
+# Combinators
+
+@docs many, some, by, spacy, between, parens, braces, brackets
+
+-}
+
+import Parser exposing (..)
+
+
+{-| Apply a parser one or more times and return a list of the results.
+-}
+some : Parser a -> Parser (List a)
```

Comment:
Having `some` return a `Parser (a, List a)` was very much intentional as to retain type safety.

Basically, `(a, List a)` represents a type-safe non-empty list which enables the use of some safe versions of `head`, `tail` or even a `foldl`/`foldr` with no need for an initial value.

If there is a use case for `some` returning only a `List a`, I guess we could implement it separately.